test/generate-upstream-cert.sh

```diff
@@ -33,8 +33,10 @@ openssl x509 -req -extfile <(printf "subjectAltName=DNS:$san") -days 365 -in "${
 
 # Create secret
 kubectl create -n ${SERVING_SYSTEM_NAMESPACE} secret generic knative-serving-certs \
-    --from-file=ca-cert.pem="${out_dir}"/root.crt --dry-run=client -o yaml | kubectl apply -f -
+    --from-file=ca-cert.pem="${out_dir}"/root.crt --dry-run=client -o yaml |  \
+    sed  '/^metadata:/a\ \ labels: {"networking.internal.knative.dev/certificate-uid":"test-id"}' | kubectl apply -f -
 
 kubectl create -n ${TEST_NAMESPACE} secret tls server-certs \
     --key="${out_dir}"/tls.key \
-    --cert="${out_dir}"/tls.crt --dry-run=client -o yaml | kubectl apply -f -
+    --cert="${out_dir}"/tls.crt --dry-run=client -o yaml | \
+    sed  '/^metadata:/a\ \ labels: {"networking.internal.knative.dev/certificate-uid":"test-id"}' | kubectl apply -f -
```

Comment:
I think we don't need to add the label to the `server-certs`. It is used by users' namespace and Kourier does not need to be reconciled when they updated them.

(If users started using a server cert signed by a different CA, Kourier needs to be reconciled to load the new CA but it is triggered by `knative-serving-certs`(=CA) change.)

Reply:
With said, the test will fail without this label? :sweat_smile:

Reply:
Kourier reconciles a related ingress referencing that secret through no? Afaik it does fail I can try once more without it.

Reply:
Yes for the CA secret (=`knative-serving-certs`) , but should be no for the server cert (=`server-certs`).
But the test failed? Hmm... I misunderstood something. I will look into it as well.

Reply:
Let me try it first. I might remember wrongly.

Reply:
Ok removed the label and switched on filtering.

Reply:
Tests passed reverting the default value for the env var.